### PR TITLE
Add log.AsYAML and log.AsYAMLDiff

### PR DESCRIPTION
**Pull request by karlkfi**
- Use the Stringer pattern (like the existing log.AsJSON) to avoid expensive diffing and serializing of objects, for logging with configurable verbosity.
- Use github.com/kylelemons/godebug/diff for diffing, because it doesn't truncate the way cmp.Diff does.
- Use the (configurable) scheme to lookup object GKV from its type and encode YAML, if possible. Fallback to yaml.Marshal for other Go types.

### CODE REVIEW

File: pkg/util/log/stringer_test.go [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+ kind: Namespace
+ metadata:
+   labels:
+-    key1: value1
++    key2: value2
+   name: example
+ `,
+		},
+		{
+			name: "core.ID (non-object compund struct) with no diff",
```
Comment (nan-yu) [inline pkg/util/log/stringer_test.go:219]: typo: compound

**Review by nan-yu — COMMENTED** [review]
one typo, otherwise lgtm

File: pkg/util/log/stringer_test.go (hunk 1 above)
Comment (karlkfi) [reply pkg/util/log/stringer_test.go:219]: fixed

Comment (nan-yu) [reply pkg/util/log/stringer_test.go:219]: line 219 and 243 also need to be fixed

Comment (karlkfi) [reply pkg/util/log/stringer_test.go]: oops. thanks. fixed.

**Review by nan-yu — APPROVED** [review]
/lgtm

**Pull request merged**